src/enable_variable_ops/ngraph_rewrite_pass.cc

```diff
@@ -249,17 +249,39 @@ class NGraphEncapsulationPass : public NGraphRewritePass {
       return Status::OK();
     }
 
+    // Get backend + its configurations
+    // to be attached to the nodes
+    std::unordered_map<std::string, std::string> config_map;
+    string backend_name = BackendManager::GetCurrentlySetBackendName();
+    const char* ng_backend_env_value = std::getenv("NGRAPH_TF_BACKEND");
+    if (ng_backend_env_value != nullptr) {
+      string backend_env = std::string(ng_backend_env_value);
+      if (backend_env.empty() ||
+          !BackendManager::IsSupportedBackend(backend_env)) {
+        return errors::Internal("NGRAPH_TF_BACKEND: ", backend_env,
+                                " is not supported");
+      }
+      backend_name = backend_env;
+    }
+    config_map = BackendManager::GetBackendAttributes(
+        backend_name);  // SplitBackendConfig
+    backend_name = config_map.at("ngraph_backend");
+    config_map.erase("ngraph_backend");
+    NGRAPH_VLOG(5) << "NGraphEncapsulationPass: backend_name " << backend_name;
+
+    // Now Process the Graph
+
     // 0. Replace optimizers then, if requested, dump the graphs.
     TF_RETURN_IF_ERROR(ReplaceModifiers(options.graph->get(), idx));
-    if (DumpMarkedGraphs()) {
+    if (DumpReplacedModifiersGraphs()) {
       DumpGraphs(options, idx, "replaced_modifier",
                  "Graph with Modifiers replaced");
     }
 
     // 1. Mark for clustering then, if requested, dump the graphs.
     std::set<string> skip_these_nodes = {};
-    TF_RETURN_IF_ERROR(
-        MarkForClustering(options.graph->get(), skip_these_nodes));
+    TF_RETURN_IF_ERROR(MarkForClustering(options.graph->get(), skip_these_nodes,
+                                         backend_name));
```

Comment:
Note to self: `MarkForClustering` and `EncapsulateClusters` are now explicitly passed the `backend`
`backend_name` is expected to be only the name `CPU` and not `CPU:0` etc.

Reply:
EncapsulateClusters are passed the additional attributes (_ngraph_device_config etc)
MarkForClustering is passed the backend name